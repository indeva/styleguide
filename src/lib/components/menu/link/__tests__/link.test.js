import React from 'react';
import renderer from 'react-test-renderer';
import Link from '..';

describe('Link', () => {
  describe('with url', () => {
    it('should render link with href', () => {
      const component = renderer.create(<Link url="/test">link with href</Link>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render link with target', () => {
      const component = renderer.create(<Link url="/test" target="_blank">link</Link>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('without url', () => {
    it('should render link properly', () => {
      const component = renderer.create(<Link>my link</Link>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
