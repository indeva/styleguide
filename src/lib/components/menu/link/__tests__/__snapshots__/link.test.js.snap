// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link with modifiers with active and child truthy 1`] = `
<span
  className="menu__link menu__link--child"
  onClick={null}
  role="presentation"
>
  test
</span>
`;

exports[`Link with modifiers with active truthy 1`] = `
<span
  className="menu__link menu__link--active"
  onClick={null}
  role="presentation"
>
  test
</span>
`;

exports[`Link with modifiers with child truthy 1`] = `
<span
  className="menu__link menu__link--child"
  onClick={null}
  role="presentation"
>
  test
</span>
`;

exports[`Link with url should render link with href 1`] = `
<a
  className="menu__link"
  href="/test"
  target="_self"
>
  link with href
</a>
`;

exports[`Link with url should render link with target 1`] = `
<a
  className="menu__link"
  href="/test"
  target="_blank"
>
  link
</a>
`;

exports[`Link without onClick prop should render link properly 1`] = `
<span
  className="menu__link"
  onClick={[Function]}
  role="presentation"
>
  my link
</span>
`;

exports[`Link without url should render link properly 1`] = `
<span
  className="menu__link"
  onClick={null}
  role="presentation"
>
  my link
</span>
`;

exports[`Link without url should render link properly with title instead of children 1`] = `
<span
  className="menu__link"
  onClick={null}
  role="presentation"
>
  my link
</span>
`;
